next-docs/pages/guides/hello-world/add-system.mdx

```diff
@@ -0,0 +1,296 @@
+
+1.  Edit `packages/client/src/mud/createSystemCalls.ts` to include `decrement`.
+    This is the file after the changes:
+
+    <CollapseCode>
+
+        ```typescript filename="createSystemCalls.ts" copy showLineNumbers {47-51,55}
+        /*
+        * Create the system calls that the client can use to ask
+        * for changes in the World state (using the System contracts).
+        */
+
+        import { getComponentValue } from "@latticexyz/recs";
+        import { ClientComponents } from "./createClientComponents";
+        import { SetupNetworkResult } from "./setupNetwork";
+        import { singletonEntity } from "@latticexyz/store-sync/recs";
+
+        export type SystemCalls = ReturnType<typeof createSystemCalls>;
+
+        export function createSystemCalls(
+          /*
+          * The parameter list informs TypeScript that:
+          *
+          * - The first parameter is expected to be a
+          *   SetupNetworkResult, as defined in setupNetwork.ts
+          *
+          * - Out of this parameter, we only care about two fields:
+          *   - worldContract (which comes from createContract, see
+          *     https://github.com/latticexyz/mud/blob/26dabb34321eedff7a43f3fcb46da4f3f5ba3708/templates/vanilla/packages/client/src/mu$
+          *   - waitForTransaction (which comes from syncToRecs, see
+          *     https://github.com/latticexyz/mud/blob/26dabb34321eedff7a43f3fcb46da4f3f5ba3708/templates/vanilla/packages/client/src/mu$
+          *
+          * - From the second parameter, which is a ClientComponent,
+          *   we only care about Counter. This parameter comes to use
+          *   through createClientComponents.ts, but it originates in
+          *   syncToRecs
+        (https://github.com/latticexyz/mud/blob/26dabb34321eedff7a43f3fcb46da4f3f5ba3708/templates/vanilla/packages/client/src/mud/setupN$
+          */
+          { worldContract, waitForTransaction }: SetupNetworkResult,
+          { Counter }: ClientComponents
+        ) {
+          const increment = async () => {
+            /*
+            * Because IncrementSystem
+            * (https://mud.dev/tutorials/walkthrough/minimal-onchain#incrementsystemsol)
+            * is in the root namespace, `.increment` can be called directly
+            * on the World contract.
+            */
+            const tx = await worldContract.write.increment();
+            await waitForTransaction(tx);
+            return getComponentValue(Counter, singletonEntity);
+          };
+
+          const decrement = async () => {
+            const tx = await worldContract.write.decrement();
+            await waitForTransaction(tx);
+            return getComponentValue(Counter, singletonEntity);
+          };
+
+          return {
+            increment,
+            decrement,
+          };
+        }
+        ```
+
+    </CollapseCode>
+
+           <details>
+
+    <summary>Explanation</summary>
+
+        {" "}
+        <p></p>
```

Comment:
do we need this?

Reply:
Weird. prettier used to break things when I committed without this, but now it does. I'll remove them.